From 105fc480b26756fc880341cf262dc9be56751e2c Mon Sep 17 00:00:00 2001
From: Daniel Engberg <daniel.engberg.lists@pyret.net>
Date: Thu, 28 Mar 2019 12:57:45 +0000
Subject: [PATCH] arm/aarch64: Set -O2 as default for Cortex-A processor cores

Platforms using Cortex-A processor cores aren't constrained in
terms of storage as much as other supported platforms such as MIPS.
Set O2 as default since Os can severly limit performance as it's
more targeted at keeping down binary size.

Tested on mvebu, sunxi

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>
Tested-by: Tad <tad@spotco.us>
---
 include/target.mk | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/target.mk b/include/target.mk
index 7526224972..b92cd7e467 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -191,7 +191,11 @@ LINUX_RECONF_DIFF = $(SCRIPT_DIR)/kconfig.pl - '>' $(call __linux_confcmd,$(filt
 ifeq ($(DUMP),1)
   BuildTarget=$(BuildTargets/DumpCurrent)
 
-  CPU_CFLAGS = -Os -pipe
+  ifneq ($(findstring cortex-a,$(CPU_TYPE)),)
+    CPU_CFLAGS = -O2 -pipe
+  else
+    CPU_CFLAGS = -Os -pipe
+  endif
   ifneq ($(findstring mips,$(ARCH)),)
     ifneq ($(findstring mips64,$(ARCH)),)
       CPU_TYPE ?= mips64
-- 
2.31.1

